packages/browser/src/index.ts

```diff
@@ -25,4 +45,33 @@ export default class LogtoClient {
   constructor(logtoConfig: LogtoConfig) {
     this.logtoConfig = logtoConfig;
   }
+
+  public async signIn(redirectUri: string) {
+    const { clientId, endpoint, requester, resources, scopes: originalScopes } = this.logtoConfig;
+    const discoveryEndpoint = new URL(discoveryPath, endpoint);
+    this.oidcConfig = await fetchOidcConfig(discoveryEndpoint.toString(), requester);
+
+    const { authorizationEndpoint } = this.oidcConfig;
+    const codeVerifier = generateCodeVerifier();
+    const codeChallenge = await generateCodeChallenge(codeVerifier);
+    const state = generateState();
+    const scopes = withReservedScopes(originalScopes).split(' ');
+
+    const signInUri = generateSignInUri({
+      authorizationEndpoint,
+      clientId,
+      redirectUri,
+      codeChallenge,
+      state,
+      scopes,
+      resources,
+    });
+
+    const logtoKey = getLogtoKey(clientId);
+    const logtoSignInSessionItem: LogtoSignInSessionItem = { redirectUri, codeVerifier, state };
```

Comment:
nit: just make it inline? 

```js
const logtoValue = JSON.stringify({ redirectUri, codeVerifier, state });
```

Reply:
intend to use type LogtoSignInSessionItem for better access when signIn and handleCallback.
better keep it？@simeng-li

Reply:
Do it in another way. PTAL.

Reply:
The source code no longer exists and I resolve this comment.